### ISSUE TITLE
Custom icon

### ISSUE THREAD

**Issue by kakawait** [florentc/xob#16]
I'd like to know if you think it could be possible to add icon or pictogram in addition to the bar?
Currently I'm using _xob_ for both light and sound system. And I'm looking to achieve something close to what _MacOS_ offers

![](https://i.stack.imgur.com/TzJRe.png)

In order to be able to differentiate (visually speaking) when sound up/down and light up/down

**Comment by florentc** [comment]
Hello,

Thank you for using xob and your suggestion. I was not planning to implement such a feature.

The style feature covers your use case (differentiating two kinds of indicators): use two instances of xob and set up two styles `[volume]` and `[light]` in your config file with different color schemes (e.g. default black and white for volume, yellow for light). Select the desired style using the `-s` option.

However, I am open to the idea and external contributions, including modest eye-candy stuff, as long as it sticks to the philosophy behind xob: simple, short, lightweight, few widespread dependencies, keeping in mind the bar may display any kind of value.

**Comment by kakawait** [comment]
Yes I'm not wondering something as fancy as MacOS but more looking for something similar to https://www.reddit.com/r/unixporn/comments/9i5seg/brightness_and_volume_notifications_with_dunst_oc/

I mean just be able to put (on configuration you could choose where to put to fit with every direction) an small icon

**Comment by florentc** [comment]
This is also what I had in mind. I would be glad to merge a pull request provided the whole thing remained light on dependencies and changes on the existing code base.

**Comment by kakawait** [comment]
@florentc I'll try to check but I'm very newbee C dev (I'm mainly Java dev) so I don't know if I'll able to produce something

**Comment by apoorv569** [comment]
> I'd like to know if you think it could be possible to add icon or pictogram in addition to the bar?
> Currently I'm using _xob_ for both light and sound system. And I'm looking to achieve something close to what _MacOS_ offers
> 
> ![](https://camo.githubusercontent.com/edea5a7b6f2a218ccb8080cbd5d7612152b35ba9/68747470733a2f2f692e737461636b2e696d6775722e636f6d2f547a4a52652e706e67)
> 
> In order to be able to differentiate (visually speaking) when sound up/down and light up/down

Can you please provide your brightness script for `xob`?

**Comment by 0Karakurt0** [comment]
I'd like to extend this enhancement a bit more: add support for unicode characters on any part of xob's "window".
And I already see config:
`text {
      content="Hello world!"
      position {
            mode=right|left|up|down|relative # in relative mode text is rendered on top of the bar
           relative_x=0.5
           relative_y=0.5 #relative to xob dimensions
       }
      style {
            colors {}
            font {}
       }
}
`
But I'll agree, this can become quite bulky

**Comment by 0Karakurt0** [comment]
Funny enough, that would solve two issues at once)

**Comment by tetchel** [comment]
Being able to render a FontAwesome icon on/next to the bar would solve this brilliantly.

**Comment by AlecsFerra** [comment]
Any news?

**Comment by SystematicError** [comment]
> Hello,
> 
> Thank you for using xob and your suggestion. I was not planning to implement such a feature.
> 
> The style feature covers your use case (differentiating two kinds of indicators): use two instances of xob and set up two styles `[volume]` and `[light]` in your config file with different color schemes (e.g. default black and white for volume, yellow for light). Select the desired style using the `-s` option.
> 
> However, I am open to the idea and external contributions, including modest eye-candy stuff, as long as it sticks to the philosophy behind xob: simple, short, lightweight, few widespread dependencies, keeping in mind the bar may display any kind of value.

While this is a viable solution, it isn't the best as multiple colors can get confusing. Personally I'd love to have all my overlays of the same colorscheme for the sake of consistency.
